montydb/database.py

```diff
@@ -56,43 +56,49 @@ def collection_names(self):
 
     def create_collection(self, name, codec_options=None, write_concern=None, **kwargs):
         """
-        Create a collection, before any insertion,
-        raise error if exists.
+        Create a collection or raise an error if it already exists.
         """
         if self.client._storage.collection_exists(self, name):
             error_msg = "collection {} already exists".format(encode_(name))
+      
             raise errors.CollectionInvalid(error_msg)
-        else:
-            collection = self.get_collection(
-                name, codec_options, write_concern, **kwargs
-            )
-            self.client._storage.collection_create(self, name)
-            return collection
+        
+        collection = self.get_collection(
+            name, codec_options, write_concern, **kwargs
+        )
+        self.client._storage.collection_create(self, name)
+       
+        return collection
 
     def drop_collection(self, name_or_collection):
-        """ """
         name = name_or_collection
+
         if isinstance(name_or_collection, MontyCollection):
             name = name_or_collection.name
         elif not isinstance(name_or_collection, string_types):
             raise TypeError("name_or_collection must be an instance of basestring")
+        
         self.client._storage.collection_drop(self, name)
 
     def get_collection(self, name, codec_options=None, write_concern=None, **kwargs):
         """
-        Get a collection.
+        Get a collection and throw an error for invalid name.
         """
-        # verify collection name
-        is_invaild = False
-        for invaild_char in ("$", "\0", "\x00"):
-            if invaild_char in name:
-                is_invaild = True
-        if name.startswith("system."):
-            is_invaild = True
+        if not name:
+            raise errors.OperationFailure("Invalid collection name - it may not be blank")
```

Comment:
Instead of collapsing errors together as one, have 3 cases where you raise the error with a specific message for the user and show the name of the collection to help debugging too 

Also invalid was spelt incorrectly